### DISCUSSION TITLE
[5주차] 네트워크 계층: 데이터 평면 4.1~5

### DISCUSSION THREAD

**Discussion by Hanjaemo** [Dijkstra-Employ/Network-Study#13] [Q&A]
5주차 네트워크 계층: 데이터 평면 4.1~5에 해당하는 질의응답입니다.

**Comment by H-sooyeon** [comment]
## Q1

IP 주소에 대해 이전에는 클래스 주소체계를 사용했습니다.
그리고 현재는 주로 CIDR이 채택되어 Subnet Mask를 사용합니다. 
클래스 주소체계에서 서브넷 마스크로 IP 주소체계를 바꾼 이유는 무엇인가요?
두 주소 체계의 차이와 함께 설명해주세요.

<br />

## Q2

IPv4의 주소공간 부족으로 인해 IPv6가 개발되었습니다.
하지만 아직은 IPv6보단 IPv4 위주로 사용하고 있습니다.
IPv4에서 바로 IPv6로 넘어가지 못하는 이유와 주소 공간이 부족함에도 아직까지 IPv4를 사용할 수 있는 이유에 대해 설명해주세요.

**Reply by Starlight258** [reply]
> ## Q1
> IP 주소에 대해 이전에는 클래스 주소체계를 사용했습니다. 그리고 현재는 주로 CIDR이 채택되어 Subnet Mask를 사용합니다. 클래스 주소체계에서 서브넷 마스크로 IP 주소체계를 바꾼 이유는 무엇인가요? 두 주소 체계의 차이와 함께 설명해주세요.
> 
`클래스 주소 체계`는 `CIDR` 채택 전 사용했던 방법으로, IP 주소의 **네트워크 부분을 `8`,`16`,`24` 비트로 제한**하고 각 서브넷을 클래스 `A`, `B`, `C` 네트워크로 분류한다.
`클래스 주소 체계`는 낭비되는 주소가 많기 때문에, 현재는 `CIDR` 방식을 채택한다.
`CIDR` 방식은 서브넷 마스크로 네트워크 부분을 나타낸다.

> ## Q2
> IPv4의 주소공간 부족으로 인해 IPv6가 개발되었습니다. 하지만 아직은 IPv6보단 IPv4 위주로 사용하고 있습니다. IPv4에서 바로 IPv6로 넘어가지 못하는 이유와 주소 공간이 부족함에도 아직까지 IPv4를 사용할 수 있는 이유에 대해 설명해주세요.

이미 `IPv4`로 전 세계의 주소체계를 구축하였기때문에, 전부 다 `IPv6`으로 옮기기엔 많은 자원이 필요하다.
또한 `IPv4`와 `IPv6 `라우터들이 `터널링`을 이용해 서로 데이터그램을 주고받을 수 있기 때문에 기존의 주소 체계를 유지하고있다.

**Comment by Starlight258** [comment]
# Q1
`SDN`에 대해 간단하게 설명해주세요.

# Q2
네트워크 계층은 `데이터 평면`과 `제어 평면`으로 나뉩니다.
각 평면에 어떤 기능이 구현되어있는지 설명해주세요.

**Reply by H-sooyeon** [reply]
> Q1 SDN에 대해 간단하게 설명해주세요.

소프트웨어 정의 네트워크(SDN)란 소프트웨어를 통해 네트워크 리소스를 가상화하고 추상화하는 네트워크 인프라에 대한 접근 방식을 의미한다.
즉, 소프트웨어 애플리케이션과 API를 이용하여 네트워크를 프로그래밍하고, 중앙에서 전체 네트워크를 제어하고 관리하는 것이다.
제어 평면의 기능을 분리된 서비스처럼 사용하여 데이터 평면과 제어 평면을 뚜렷하게 분리한다.
제어 평면에서 여러 네트워크 장비를 제어하기 때문에 관리가 간소화되고 운영에 들어가는 비용을 줄일 수 있으며 각 장비의 사양을 각각의 기능에 최적화시킬 수 있다는 장점이 있다.

<br />

> Q2 네트워크 계층은 데이터 평면과 제어 평면으로 나뉩니다. 각 평면에 어떤 기능이 구현되어있는지 설명해주세요.

각 라우터의 데이터 평면 역할은 입력 링크에서 출력 링크로 데이터그램을 전달하는 것이다.
데이터 평면은 패킷 전달 및 처리, 트래픽 관리를 수행한다.
제어 평면의 근본적인 역할은 데이터그램 출발지 호스트에서 목적지 호스트까지 잘 전달되게끔 로컬 포워딩, 라우터별 포워딩을 대응시키는 것이다.
제어 평면의 주요 기능은 라우팅으로, 네트워크 장치가 최적의 경로를 찾아 패킷을 전송할 수 있도록 라우팅 테이블을 설정한다.

**Reply by Starlight258** [reply]
> # Q1
> `SDN`에 대해 간단하게 설명해주세요.
> 
`소프트웨어 기반 네트워크`의 약자로, `원격 컨트롤러`를 통해 테이블이 계산, 설치 및 업데이트된다.
기존의 라우터는 제어부와 전송부가 같이 존재했지만, `SDN`을 이용해 제어부와 전송부를 분리하고 제어부를 별도의 컴퓨팅 서버에 맡긴다.
`제어부`를 통해 여러 네트워크 장비를 제어할 수 있어 관리가 간소화되어 비용을 절감할 수 있고, 
네트워크를 `가상화`하여 원하는 시기에 필요한 만큼 리소스를 확장하거나 축소할 수 있다.

> # Q2
> 네트워크 계층은 `데이터 평면`과 `제어 평면`으로 나뉩니다. 각 평면에 어떤 기능이 구현되어있는지 설명해주세요.
- `데이터 평면` : 입력 링크에서 출력 링크로 **데이터그램을 전달** 하는 기능
  - `IP 포워딩`과 `일반화된 포워딩`
- `제어 평면` : 데이터그램이 송신 호스트에서 목적지 호스트까지 전달되게끔 **로컬, 퍼 라우터 포워딩 조정**하는 기능 
  - `OSPF`, `BGP` 라우팅 프로토콜, `SDN`

**Comment by Hanjaemo** [comment]
### Q1  
CIDR의 등장 배경에 대해 설명해 주세요.

### Q2
DHCP는 IP 주소 외의 다른 정보를 추가로 전달할 수 있습니다. 이때 서브넷 마스크도 함께 전달하는데, 그 이유는 무엇일까요?

**Reply by Starlight258** [reply]
> ### Q1
> CIDR의 등장 배경에 대해 설명해 주세요.
> 
기존에 수행되고 있던 `클래스 주소 체계`는 낭비되는 주소가 많기 때문에, 현재는 `CIDR` 방식을 채택한다.
CIDR 방식은 서브넷 마스크로 네트워크 부분을 나타낸다.

> ### Q2
> DHCP는 IP 주소 외의 다른 정보를 추가로 전달할 수 있습니다. 이때 서브넷 마스크도 함께 전달하는데, 그 이유는 무엇일까요?
- 서브넷 마스크를 통해 장치는 자신이 속한 네트워크를 정확히 식별할 수 있다.

**Comment by choiyoung69** [comment]
### Q1
스위칭 구조에서는 여러 가지 방법으로 스위칭을 수행할 수 있습니다.. 각 방법에 대한 특징에 대해 간단하게 설명해주세요

### Q2
NAT 기능은 최근 폭넓게 쓰이지만, 문제가 없는 것은 아닙니다. NAT 기능의 단점에 대해 '포트 번호'를 포함하여 설명해주세요.
